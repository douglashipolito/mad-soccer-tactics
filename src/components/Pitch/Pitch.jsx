import React from "react";
import "./Pitch.scss";

class Pitch extends React.Component {
  render() {
    return (
      <div className="football-pitch">
        {this.props.children}
        <div className="outline marking" />
        <div className="box left marking" />
        <div className="box-d left marking" />
        <div className="box left small marking" />
        <div className="box right marking" />
        <div className="box-d right marking" />
        <div className="box right small marking" />
        <div className="spot left marking" />
        <div className="spot right marking" />
        <div className="spot center marking" />
        <div className="center-line marking" />
        <div className="center-circle marking" />
        <div className="corner top left marking" />
        <div className="corner top right marking" />
        <div className="corner bottom left marking" />
        <div className="corner bottom right marking" />
        <div className="grass" />
      </div>
    );
  }
}

export default Pitch;
