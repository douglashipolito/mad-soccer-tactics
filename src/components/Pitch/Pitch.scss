.football-pitch {
  $color: #c35b3b;
  $stripeWidth: 60px;
  $markingWidth: 4px;
  $markingColor: white;
  $stripeStrength: 0.1;
  $width: 100%;
  $height: 780px;

  box-shadow: -2px 4px 6px 1px rgba(0, 0, 0, 0.5);
  width: $width;
  height: $height;
  position: relative;
  overflow: hidden;

  .marking {
    background: $markingColor;
    position: absolute;
    z-index: 2;
  }

  .outline.marking {
    border: $markingWidth solid $markingColor;
    width: 100%;
    height: 100%;
    background: transparent;
  }

  .center-line {
    width: $markingWidth;
    height: 100%;
    top: 0;
    left: calc(50% - #{$markingWidth} / 2);
  }

  .center-circle {
    $size: 120px;
    width: $size;
    height: $size;
    border-radius: $size;
    border: $markingWidth solid $markingColor;
    background: none;
    top: calc(50% - #{$size} / 2);
    left: calc(50% - #{$size} / 2);
  }

  .box {
    top: 25%;
    width: 16%;
    height: 50%;
    background: transparent;
    border: $markingWidth solid $markingColor;

    &.left {
      left: 0;
    }

    &.right {
      right: 0;
    }

    &.small {
      top: 36%;
      width: 8%;
      height: 28%;
    }
  }

  .box-d {
    top: 35%;
    width: 15%;
    height: 30%;
    background: transparent;
    border: $markingWidth solid $markingColor;

    &.left {
      left: 6%;
      border-left: none;
      border-bottom-right-radius: 1000px;
      border-top-right-radius: 1000px;
      clip-path: inset(0 0 0 70%);
    }

    &.right {
      right: 6%;
      border-right: none;
      border-bottom-left-radius: 1000px;
      border-top-left-radius: 1000px;
      clip-path: inset(0 70% 0 0);
    }
  }

  .spot {
    width: $markingWidth;
    height: $markingWidth;
    border-radius: $markingWidth;
    top: calc(50% - #{$markingWidth} / 2);

    &.left {
      left: 11%;
    }

    &.right {
      right: 11%;
    }

    &.center {
      width: $markingWidth * 2;
      height: $markingWidth * 2;
      left: calc(50% - #{$markingWidth});
    }
  }

  .corner {
    $size: 40px;
    width: $size;
    height: $size;
    border-radius: $size;
    background: none;
    border: $markingWidth solid $markingColor;

    &.top {
      top: -$size/2;
    }

    &.left {
      left: -$size/2;
    }

    &.bottom {
      bottom: -$size/2;
    }

    &.right {
      right: -$size/2;
    }
  }

  .grass {
    width: 100%;
    height: 100%;
    background-color: $color;
  }

  .grass:before {
    content: "";
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 2;
  }

  .grass:after {
    content: "";
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 1;
    background: repeating-linear-gradient(
      to right,
      rgba(0, 0, 0, $stripeStrength),
      rgba(0, 0, 0, $stripeStrength) $stripeWidth,
      transparent $stripeWidth,
      transparent $stripeWidth * 2
    );
  }
}
