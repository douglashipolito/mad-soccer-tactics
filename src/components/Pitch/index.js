import Pitch from "./Pitch.jsx";

export default Pitch;
