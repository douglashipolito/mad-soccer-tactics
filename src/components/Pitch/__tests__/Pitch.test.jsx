import React from "react";
import { create } from "react-test-renderer";
import Pitch from "../";

test("Pitch Renders", () => {
  const string = create(<Pitch />);
  expect(string).toMatchSnapshot();
});
