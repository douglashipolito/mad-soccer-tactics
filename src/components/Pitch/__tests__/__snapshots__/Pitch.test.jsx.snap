// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pitch Renders 1`] = `
<div
  className="football-pitch"
>
  <div
    className="outline marking"
  />
  <div
    className="box left marking"
  />
  <div
    className="box-d left marking"
  />
  <div
    className="box left small marking"
  />
  <div
    className="box right marking"
  />
  <div
    className="box-d right marking"
  />
  <div
    className="box right small marking"
  />
  <div
    className="spot left marking"
  />
  <div
    className="spot right marking"
  />
  <div
    className="spot center marking"
  />
  <div
    className="center-line marking"
  />
  <div
    className="center-circle marking"
  />
  <div
    className="corner top left marking"
  />
  <div
    className="corner top right marking"
  />
  <div
    className="corner bottom left marking"
  />
  <div
    className="corner bottom right marking"
  />
  <div
    className="grass"
  />
</div>
`;
